runtime/lua/vim/lsp/buf.lua

```diff
@@ -571,35 +581,48 @@ end
 --- Requests code actions from all clients and calls the handler exactly once
 --- with all aggregated results
 ---@private
-local function code_action_request(params)
+local function code_action_request(params, opts)
   local bufnr = vim.api.nvim_get_current_buf()
   local method = 'textDocument/codeAction'
   vim.lsp.buf_request_all(bufnr, method, params, function(results)
-    on_code_action_results(results, { bufnr = bufnr, method = method, params = params })
+    ctx = { bufnr = bufnr, method = method, params = params, opts = opts}
+    on_code_action_results(results, ctx)
```

Comment:
the `ctx` is also passed along to client side commands, I'm not sure if it's good to add extra state that is then leaked out.